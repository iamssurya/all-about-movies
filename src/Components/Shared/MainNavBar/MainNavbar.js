import React from "react";
import { Col, Form, FormControl, Navbar, Row } from "react-bootstrap";
import { connect } from "react-redux";
import { peekSearchMovies, peekMovies } from "store/actions/movies";
import { DebounceInput } from "react-debounce-input";
import { selectMovieStore } from "store/dispatchers";
import { MdMovieCreation } from "react-icons/md";
import { withRouter } from "react-router-dom";

const DEFAULT_SEARCH = "popular";

class MainNavBar extends React.Component {
  searchMovies = (event) => {
    event.preventDefault();
    const { dispatch } = this.props;
    if (event.target.value) {
      dispatch(selectMovieStore(event.target.value));
      dispatch(peekSearchMovies(event.target.value));
    } else {
      dispatch(selectMovieStore(DEFAULT_SEARCH));
      dispatch(peekMovies(DEFAULT_SEARCH));
    }
  };

  goToHome = () => {
    this.props.history.push("/");
  };

  render() {
    return (
      <React.Fragment>
        <Navbar bg="dark" variant="dark" fixed="top">
          <Row className="w-100">
            <Col>
              <Navbar.Brand onClick={this.goToHome}>
                <h4>
                  <MdMovieCreation />
                </h4>
              </Navbar.Brand>
            </Col>
            <Col xs={4}>
              <Form inline>
                <FormControl
                  as={DebounceInput}
                  type="text"
                  placeholder="Search Movies..."
                  debounceTimeout={1000}
                  className="mr-sm-2 w-100"
                  onChange={this.searchMovies}
                />
              </Form>
            </Col>
          </Row>
        </Navbar>
      </React.Fragment>
    );
  }
}

export default React.memo(withRouter(connect()(MainNavBar)));
