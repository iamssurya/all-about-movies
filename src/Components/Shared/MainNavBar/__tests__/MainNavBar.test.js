import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import MainNavbar from "../MainNavBar";

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureMockStore();
const store = mockStore({});

describe("<MainNavBar /> component tests", () => {
  let Component;
  let Wrapper;
  beforeEach(() => {
    Component = mount(
      <Provider store={store}>
        <MainNavbar />
      </Provider>
    );
    Wrapper = toJson(Component);
  });

  it("Should render component correctly", () => {
    expect(Wrapper).toMatchSnapshot();
  });
});
