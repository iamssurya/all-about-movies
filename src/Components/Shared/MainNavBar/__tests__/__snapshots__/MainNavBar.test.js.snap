// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNavBar /> component tests Should render component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(MainNavBar)>
    <MainNavBar
      dispatch={[Function]}
    >
      <Navbar
        bg="dark"
        collapseOnSelect={false}
        expand={true}
        fixed="top"
        variant="dark"
      >
        <nav
          className="navbar navbar-expand navbar-dark bg-dark fixed-top"
        >
          <Row
            className="w-100"
            noGutters={false}
          >
            <div
              className="w-100 row"
            >
              <Col>
                <div
                  className="col"
                >
                  <NavbarBrand
                    href="/"
                  >
                    <a
                      className="navbar-brand"
                      href="/"
                    >
                      <h4>
                        <MdMovieCreation>
                          <IconBase
                            attr={
                              Object {
                                "viewBox": "0 0 24 24",
                              }
                            }
                          >
                            <svg
                              fill="currentColor"
                              height="1em"
                              stroke="currentColor"
                              strokeWidth="0"
                              style={
                                Object {
                                  "color": undefined,
                                }
                              }
                              viewBox="0 0 24 24"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
                                key="0"
                              />
                            </svg>
                          </IconBase>
                        </MdMovieCreation>
                      </h4>
                    </a>
                  </NavbarBrand>
                </div>
              </Col>
              <Col
                xs={4}
              >
                <div
                  className="col-4"
                >
                  <Form
                    inline={true}
                  >
                    <form
                      className="form-inline"
                    >
                      <FormControl
                        as={[Function]}
                        className="mr-sm-2 w-100"
                        debounceTimeout={1000}
                        onChange={[Function]}
                        placeholder="Search Movies..."
                        type="text"
                      >
                        <DebounceInput
                          className="mr-sm-2 w-100 form-control"
                          debounceTimeout={1000}
                          element="input"
                          forceNotifyByEnter={true}
                          forceNotifyOnBlur={true}
                          minLength={0}
                          onChange={[Function]}
                          placeholder="Search Movies..."
                          type="text"
                        >
                          <input
                            className="mr-sm-2 w-100 form-control"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            placeholder="Search Movies..."
                            type="text"
                            value=""
                          />
                        </DebounceInput>
                      </FormControl>
                    </form>
                  </Form>
                </div>
              </Col>
            </div>
          </Row>
        </nav>
      </Navbar>
    </MainNavBar>
  </Connect(MainNavBar)>
</Provider>
`;
